### PR TITLE
Feat/trace db update from server

### PR DESCRIPTION
**Pull request by aadnekar**

### CODE REVIEW

**Review by datenreisender — APPROVED** [review]
Overall the PR looks fine. 

Just two things where I am not sure whether they are correct and you should take a look at:

- You added the files `resources/traceDB/mfw_nrf9160_1.3.5_trace-db.json` and `resources/traceDB/mfw_nrf9160_1.3.5-FOTA-TEST_trace-db.json`, but did not also add them to `resources/traceDB/config.json`. Is that correct?
- When downloading a `config.json` from the server, it is not stored locally. Only the DB files are stored. Doesn’t modem-lib also need an updated `config.json`, e.g. to detect a new modem DB version?

_All_ other comments, which I added to individual code places, are just suggestions that should not block merging this. You can also merge this first and then consider later whether you want to incorporate the suggestions. Or, if you like, we can also take a look at them together.

One more suggestion: I think the whole code to download the database files is long enough that, instead of being added to `traceDatabase.ts`, it warrants being put in a source file on its own.

File: src/features/tracing/traceDatabase.ts [hunk 1]
```diff
@@ -55,3 +77,71 @@ export const setSelectedTraceDatabaseFromVersion =
...
+                      err
+                  );
+                  return localDatabasesCache;
+              });
+
+const downloadAll = async (config: TraceConfig) => {
+    await prepareTargetDirectory();
+
+    const databases =
+        config.firmwares.devices[0].databases.reverse() as Database[];
```
[the remark this answers is not in the document]
Comment (aadnekar) [reply src/features/tracing/traceDatabase.ts:115]: We use `reverse()` in order to list the latest versions first. The cast is no longer needed as I added the `TraceConfig` type 👍

File: src/features/tracing/traceDatabase.ts [hunk 2]
```diff
@@ -14,28 +16,48 @@ import {
...
     uuid: string;
     version: string;
 };
 
-let databasesCache: Database[];
+const SERVER_URL =
+    'https://developer.nordicsemi.com/.pc-tools/nrfconnect-apps/trace-db';
+
+// Fallback value added in order to make tests pass.
+// If fallback value, trace files will end up in pc-nrfconnect-launcher
```
[the remark this answers is not in the document]
Comment (aadnekar) [reply src/features/tracing/traceDatabase.ts:40]: Forgot to delete the comment after I changed my code. Thanks!

File: src/features/tracing/traceDatabase.ts [hunk 3]
```diff
@@ -55,3 +77,71 @@ export const setSelectedTraceDatabaseFromVersion =
...
+                  } catch (err) {
+                      logger.error(`Could not parse config.json: ${err}`);
+                      return [];
+                  }
+
+                  if (!config) return [];
+
+                  await downloadAll(config);
+
+                  remoteDatabasesCache = config.firmwares.devices[0].databases;
```
[the remark this answers is not in the document]
Comment (aadnekar) [reply src/features/tracing/traceDatabase.ts:100]: Extracted to a task I want to do separately. We should also make some changes to the config.json structure, but I would like to spend some more time thinking through the changes before introducing them.

File: src/features/tracing/traceDatabase.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (datenreisender) [reply src/features/tracing/traceDatabase.ts]: Ah, yes, I am seeing now that this is also done for the local databases further up. Maybe just do it once and closer to where it is actually needed (in the React component), making it more obvious what it is needed for. E.g. change https://github.com/NordicSemiconductor/pc-nrfconnect-cellularmonitor/blob/96757bbda7cac7b40f32fbc522a658746330eb23/src/components/SidePanel/DatabaseFileOverride.tsx#L54 to `databases.map(…).reverse()`

**Pull request merged**